plugins/in_splunk/splunk_prot.c

```diff
@@ -0,0 +1,822 @@
+
+static inline int mk_http_point_header(mk_ptr_t *h,
+                                       struct mk_http_parser *parser, int key)
+{
+    struct mk_http_header *header;
+
+    header = &parser->headers[key];
+    if (header->type == key) {
+        h->data = header->val.data;
+        h->len  = header->val.len;
+        return 0;
+    }
+    else {
+        h->data = NULL;
+        h->len  = -1;
+    }
+
+    return -1;
+}
+
+/*
+ * Handle an incoming request. It perform extra checks over the request, if
+ * everything is OK, it enqueue the incoming payload.
+ */
+int splunk_prot_handle(struct flb_splunk *ctx, struct splunk_conn *conn,
+                       struct mk_http_session *session,
+                       struct mk_http_request *request)
+{
+    int i;
+    int ret;
+    int len;
+    char *uri;
+    char *qs;
+    off_t diff;
+    flb_sds_t tag;
+    struct mk_http_header *header;
+
+    if (request->uri.data[0] != '/') {
+        send_response(conn, 400, "error: invalid request\n");
+        return -1;
+    }
+
+    /* Decode URI */
+    uri = mk_utils_url_decode(request->uri);
+    if (!uri) {
+        uri = mk_mem_alloc_z(request->uri.len + 1);
+        if (!uri) {
+            return -1;
+        }
+        memcpy(uri, request->uri.data, request->uri.len);
+        uri[request->uri.len] = '\0';
+    }
+
+    /* Try to match a query string so we can remove it */
+    qs = strchr(uri, '?');
+    if (qs) {
+        /* remove the query string part */
+        diff = qs - uri;
+        uri[diff] = '\0';
+    }
+
+    /* Refer the tag at first*/
+    if (ctx->ins->tag && !ctx->ins->tag_default) {
+        tag = flb_sds_create_size(1024);
+        if (!tag) {
+            return -1;
+        }
+        /* clear tag before using */
+        flb_sds_len_set(tag, 0);
+        flb_sds_cat(tag, ctx->ins->tag, flb_sds_len(ctx->ins->tag));
+    }
+    else {
+        /* Compose the query string using the URI */
+        len = strlen(uri);
+
+        if (len == 1) {
+            tag = NULL; /* use default tag */
+        }
+        else {
+            tag = flb_sds_create_size(len);
+            if (!tag) {
+                mk_mem_free(uri);
+                return -1;
+            }
+
+            /* New tag skipping the URI '/' */
+            flb_sds_cat(tag, uri + 1, len - 1);
+
+            /* Sanitize, only allow alphanum chars */
+            for (i = 0; i < flb_sds_len(tag); i++) {
+                if (!isalnum(tag[i]) && tag[i] != '_' && tag[i] != '.') {
+                    tag[i] = '_';
+                }
+            }
+        }
+    }
+
+    /* Check if we have a Host header: Hostname ; port */
+    mk_http_point_header(&request->host, &session->parser, MK_HEADER_HOST);
+
+    /* Header: Connection */
+    mk_http_point_header(&request->connection, &session->parser,
+                         MK_HEADER_CONNECTION);
+
+    /* HTTP/1.1 needs Host header */
+    if (!request->host.data && request->protocol == MK_HTTP_PROTOCOL_11) {
```

Comment:
`request->host.data == NULL`

Reply:
Fixed in https://github.com/fluent/fluent-bit/pull/7349/commits/bca2d3ceb653f6f75405d4232dc87af5824d8736.